core/src/com/unciv/logic/automation/unit/WorkerAutomation.kt

```diff
@@ -256,9 +281,11 @@ class WorkerAutomation(
         val workableTiles = currentTile.getTilesInDistance(4)
                 .filter {
                     (it.civilianUnit == null || it == currentTile)
-                            && tileCanBeImproved(unit, it)
-                            && it.getTilesInDistance(2)
-                            .none { tile -> tile.isCityCenter() && tile.getCity()!!.civInfo.isAtWarWith(civInfo) }
+                            && (tileCanBeImproved(unit, it) || it.isPillaged())
```

Comment:
Shouldn't "tileCanBeImproved" just automatically return True if the tile is pillaged and the unit can repair pillaged tiles?

Reply:
tileCanBeImproved seems mostly concerned about building a new improvement and not handling Repair. Fixing the logic to enforce this